src/metabase/db/liquibase.clj

```diff
@@ -200,13 +200,16 @@
       (catch Exception e
         (log/error e (trs "Unable to release the Liquibase lock after a migration failure"))))))
 
-(defn- wait-for-migration-lock-to-be-cleared
+(defn- lock-service ^LockService [^Liquibase liquibase]
+  (.getLockService (LockServiceFactory/getInstance) (.getDatabase liquibase)))
+
+(defn- wait-for-migration-lock
   "Check and make sure the database isn't locked. If it is, sleep for 2 seconds and then retry several times. There's a
   chance the lock will end up clearing up so we can run migrations normally."
   [^Liquibase liquibase]
   (let [retry-counter (volatile! 0)]
     (u/auto-retry 5
-      (when (migration-lock-exists? liquibase)
+      (when-not (.acquireLock (lock-service liquibase))
```

Comment:
why not keep using `migration-lock-exists?`?

Reply:
The semantics are different - this checks whether the given liquibase instance is holding the lock, versus it potentially being held by another thread or process.

Reply:
This method already encapsulates checking whether it is taken, and immediately returns false if it is. We don't want to call `waitForLock`, because that could trigger an additional pause, which by default is actually in minutes not seconds.